build.sbt

```diff
@@ -13,131 +12,191 @@ lazy val spire = project.in(file("."))
+  .settings(spireSettings:_*)
+  .settings(libraryDependencies ++= Seq(
+    "org.typelevel" %%% "discipline" % "0.4",
+    "org.scalacheck" %%% "scalacheck" % "1.12.4"
+  ))
+  .jvmSettings(scalacOptions += "-optimize")
   .dependsOn(core)
 
-lazy val tests = project
+lazy val lawsJVM = laws.jvm
+lazy val lawsJS = laws.js
+
+// Todo: As all tests in this list are commented out, no tests in testJS are run - but they are compiled.
+lazy val jsTests = List(
+ /* "spire.PartialOrderSyntaxTest",
+  "spire.PartialSyntaxTest",
+  "spire.SyntaxTest",
+  //"spire.algebra.GCDTest",
+  "spire.algebra.NRootTest",
+  "spire.algebra.PartialOrderTest",
+  "spire.algebra.RingTest",
+  "spire.algebra.SignedTest",
+  "spire.algebra.TrigTest",
+  "spire.laws.LawTests",
+  "spire.math.AlgebraicTest",
+  "spire.math.BinaryMergeCheck",
+  "spire.math.BitStringCheck",
+  "spire.math.BitStringTest",
+  "spire.math.ComplexCheck",
+  "spire.math.ComplexCheck2",
+  "spire.math.ComplexTest",
+  "spire.math.ContinuousIntervalTest",
+  "spire.math.CooperativeEqualityTest",
+  "spire.math.FastComplexCheck",
+  "spire.math.FixedPointCheck",
+  "spire.math.FpFilterTest",
+  "spire.math.IntervalCheck",
+  "spire.math.IntervalGeometricPartialOrderTest",
+  "spire.math.IntervalIteratorCheck",
+  "spire.math.IntervalReciprocalTest",
+  "spire.math.IntervalSubsetPartialOrderTest",
+  "spire.math.IntervalTest",
+  "spire.math.JetTest",
+  "spire.math.LinearSelectTest",
+  "spire.math.LiteralsTest",
+  "spire.math.MergingTest",
+  "spire.math.NaturalTest",
+  "spire.math.NumberPropertiesTest",
+  "spire.math.NumberTest",
+  "spire.math.NumericTest",
+  "spire.math.PackageCheck",
+  "spire.math.PackageTest",
+  "spire.math.PolynomialCheck",
+  "spire.math.PolynomialSamplingCheck",
+  "spire.math.PolynomialTest",
+  "spire.math.QuaternionCheck",
+  "spire.math.QuickSelectTest",
+  "spire.math.RationalCheck",
+  "spire.math.RationalTest",
+  "spire.math.RealCheck",
+  "spire.math.RingIntervalTest",
+  "spire.math.SafeLongTest",
+  "spire.math.SearchTest",
+  "spire.math.SortingTest",
+  "spire.math.TrileanCheck",
+  "spire.math.UByteTest",
+  "spire.math.UIntTest",
+  "spire.math.ULongTest",
+  "spire.math.UShortTest",
+  "spire.math.prime.FactorHeapCheck",
+  "spire.math.prime.FactorsCheck",
+  "spire.random.GaussianTest",      
+  "spire.random.GeneratorTest",
+  "spire.random.SamplingTest",
+  "spire.random.ShufflingTest",
+  "spire.syntax.CforTest",
+  "spire.util.OptCheck",
+  "spire.util.PackCheck",
+  "test.scala.spire.math.TypeclassExistenceTest"
+*/
+)
+
+lazy val tests = crossProject.crossType(CrossType.Pure)
   .settings(moduleName := "spire-tests")
-  .settings(spireSettings)
-  .settings(testsSettings)
-  .settings(noPublishSettings)
+  .settings(spireSettings:_*)
+  .settings(scalaTestSettings:_*)
+  .settings(noPublishSettings:_*)
+  .jvmSettings(scalacOptions += "-optimize")
+  .jsSettings(testOptions in Test := Seq(Tests.Filter(s => jsTests.contains(s))))
```

Comment:
I can do, but the biggest issue is with tests that do not return - so you don't actually know what is happening. So the idea here is that I can test them one by one (yawn). Then, it makes sense to filter out the bad ones, sure